### PR TITLE
implement appfile oriented multiple OAM Application orchestration

### LINKED ISSUE #20: [feature] Support Appset to bundle your apps
appset init will create a directory

```
$ rudr appset:init myapp --tags env=prod,team=MyTeam --domain example.com
Creating at  ~/rudr/myapp/rudr.cue
SUCCESS!

$ rudr deployment:init frontend -p 80 --image resouer/demo:v1
Adding component frontend to myapp
Updating ~/rudr/myapp/frontend.cue
SUCCESS!

$ rudr route frontend --domian frontend.oam.dev
Binding route --- frontend
Updating ~/rudr/myapp/frontend.cue
SUCCESS!

$ rudr run frontend
Deploying frontend
Updating ~/rudr/myapp/rudr.cue
SUCCESS!

$ rudr delete frontend
```

### LINKED ISSUE #108: [Feature] vela <workload>:run --stage
For apps draft,  stage, and update/deploy later.

### PR DESCRIPTION
**Pull request by wonderflow**
- [x] 1. refactor workload:run
- [x] 2. refactor trait attach/detach
- [x] 3. support vela app:run <appname>

fixes #108 
fixes #20

### CODE REVIEW

File: pkg/cmd/env.go [hunk 1]
```diff
@@ -59,7 +59,7 @@ func NewEnvInitCommand(c types.Args, ioStreams cmdutil.IOStreams) *cobra.Command
 	var envArgs types.EnvMeta
 	ctx := context.Background()
 	cmd := &cobra.Command{
-		Use:                   "env:init",
+		Use:                   "env:init <envName>",
```
Comment (zzxwill) [inline pkg/cmd/env.go:62]: If `--namespace` is not set, `default` namespace will be created for the env?

File: pkg/cmd/show.go [hunk 2]
```diff
@@ -18,14 +18,15 @@ import (
 func NewAppShowCommand(c types.Args, ioStreams cmdutil.IOStreams) *cobra.Command {
 	ctx := context.Background()
 	cmd := &cobra.Command{
-		Use:     "app:show",
+		Use:     "app:show <APPLICATION-NAME>",
+		Aliases: []string{"show"},
```
Comment (zzxwill) [inline pkg/cmd/show.go:22]: Nice.

File: pkg/cmd/show.go [hunk 3]
```diff
@@ -18,14 +18,15 @@ import (
...
 		Short:   "get detail spec of your app",
 		Long:    "get detail spec of your app, including its workload and trait",
 		Example: `vela app:show <APPLICATION-NAME>`,
```
Comment (zzxwill) [inline pkg/cmd/show.go:25]:
```suggestion
		Example: `vela app:show <APPLICATION_NAME>`,
```

File: pkg/cmd/status.go [hunk 4]
```diff
@@ -24,7 +24,8 @@ type ApplicationStatusMeta struct {
 func NewAppStatusCommand(c types.Args, ioStreams cmdutil.IOStreams) *cobra.Command {
 	ctx := context.Background()
 	cmd := &cobra.Command{
-		Use:     "app:status",
+		Use:     "app:status <APPLICATION-NAME>",
```
Comment (zzxwill) [inline pkg/cmd/status.go:27]:
```suggestion
		Use:     "app:status <APPLICATION_NAME>",
```

File: pkg/cmd/trait.go [hunk 5]
```diff
@@ -206,48 +168,36 @@ func AddTraitDetachCommands(parentCmd *cobra.Command, c types.Args, ioStreams cm
...
 
 func (o *commandOptions) Run(cmd *cobra.Command, ctx context.Context) error {
 	if o.Detach {
-		o.Info("Detaching trait from app", o.Component.Name)
+		o.Infof("Detaching %s from app %s\n", o.Template.Name, o.workloadName)
 	} else {
-		o.Info("Applying trait for app", o.Component.Name)
+		o.Infof("Adding %s for app %s \n", o.Template.Name, o.workloadName)
+	}
+	staging, err := strconv.ParseBool(cmd.Flag(Staging).Value.String())
```
Comment (zzxwill) [inline pkg/cmd/trait.go:192]: Bool flag can be directly retrieved in this way `staging, err := cmd.Flags().GetBool(Staging)`.

Comment (wonderflow) [reply pkg/cmd/trait.go:192]: cool

File: pkg/cmd/env.go (hunk 1 above)
Comment (wonderflow) [reply pkg/cmd/env.go:62]: yes

**Pull request merged**